### PR TITLE
Fixes Grid Check Event creating challenging engine issue

### LINKED ISSUE #27902: Power outage event STILL MAKES THE SM DELAMINATE
This issue persists on MetaStation, that fix PR didn't fix shit.
( round id 71195 if needed )
Basically, the power outage event requires IMMEDIATE attention to SM.
If you don't instantly go there, turn the APC on and everything else? It will DELAMINATE.
Hell, it's hard enough just to do that! The SM begins spawning electricity arcs less than a minute after the power outage, knocking down people.

### PR DESCRIPTION
**Pull request by Penguaro**
Fixes #27902

### CODE REVIEW

**Comment by Shadowlight213** [comment]
There's no need for this. A single critical machine in an area, makes the entire area be spared.

**Comment by Penguaro** [comment]
@ShizCalev I can't speak for the solution you provided one way or another but previously this variable was created to prevent people in the middle of cloning from dying during the Grid Check event. Flagging the Variable Critical Machine will make that Area unaffected by the event.

**Comment by Penguaro** [comment]
@Shadowlight213 was there one specific filter you want the variable flagged with? I thought it best to do all of them in case of future changes to the layout and this being overlooked.

**Comment by Penguaro** [comment]
And currently, no devices in the area have the flag since SM replaced the Particle Accelerator.

**Comment by Shadowlight213** [comment]
Would adding the flag to the sm itself work? As long as it's not in a new area, it should work.
I'd need to take a look, but maybe make the atmos alarm in the area critical or something if just the sm doesn't work

**Comment by Penguaro** [comment]
The SM itself is in a different area. The problem isn't the SM, AirAlarm or Vents, it is the Filters for the loop.

**Comment by Penguaro** [comment]
As far as I know, this variable only affects Grid Check, what is the concern with having the filters in that room var edited for this flag? The filters in Atmospherics are untouched.

**Comment by Shadowlight213** [comment]
No concern at all really. it just seems silly to have all of then edited. I'll probably look into changing the existing code a bit to be more sm friendly

**Comment by Penguaro** [comment]
You're more savvy with DM than I. Do you know why ShizCalev's solution in #27737 didn't work?

**Comment by ShizCalev** [comment]
Aren't there like 6 different power events? I probably only hit one of the checks with #27737

**Comment by Penguaro** [comment]
I've only ever seen the one to be honest. What others are there besides Grid Check?

**Comment by Shadowlight213** [comment]
None

**Comment by Fox-McCloud** [comment]
If a machine is to be critical, don't map edit it; it should be in the code itself.

This is a horrendous way of handling it because it hides game balance in map.

**Comment by Penguaro** [comment]
@Fox-McCloud Editing the Filters in the code would make all of them "Critical" and thus make many areas that use them stay powered during Grid Check. The variable is only used for that event so far.

**Comment by Fox-McCloud** [comment]
Then you should make a subtype that has that var set and explicitly make a comment in the code what it's used for and why; again, balance changes and types like this should be explicitly set when possible, as it makes it easier for both coders *and* mappers in the future.

**Comment by Penguaro** [comment]
Well, today I learned that even though the cloner is one of the objects the Critical Object variable was created for, it was never actually used for the cloner. So that begs the question, should a sub-type be created for the cloner as well?

Thoughts? @Shadowlight213 @Fox-McCloud

**Comment by Penguaro** [comment]
@Fox-McCloud Made the changes you suggested
@ShizCalev What you did in the other PR was a fix for when an Admin triggers Make All Areas Unpowered event under Secrets.

**Comment by Penguaro** [comment]
@MMMiracles FYI as this makes a change on Cere.

File: code/modules/atmospherics/machinery/components/trinary_devices/filter.dm [hunk 1]
```diff
@@ -12,6 +12,14 @@
 /obj/machinery/atmospherics/components/trinary/filter/flipped
 	icon_state = "filter_off_f"
 	flipped = 1
+	
+// These two filter types have critical_machine flagged to on and thus causes the area they are in to be exempt from the Grid Check event.
+	
+/obj/machinery/atmospherics/components/trinary/filter/critical
+	critical_machine = 1
```
Comment (lzimann) [inline code/modules/atmospherics/machinery/components/trinary_devices/filter.dm]: `TRUE` please

Comment (Penguaro) [inline code/modules/atmospherics/machinery/components/trinary_devices/filter.dm]: @lzimann I'm happy to change it to True but by the default it shows as a 0. Will TRUE be a problem?

**Comment by Penguaro** [comment]
@lzimann Is there a distinction for when to use TRUE/FALSE vs 1/0 or are we as a codebase just moving toward TRUE/FALSE? I ask so I can keep that in mind as I make future PRs and want to understand why you requested the code change. I'm not criticizing, just asking.

**Comment by lzimann** [comment]
Whenever it's a bool(and will be used as one) you use TRUE/FALSE, otherwise just the number.

**Pull request merged**